includes/Classifai/Command/ClassifaiCommand.php

```diff
@@ -365,6 +366,188 @@ public function text_to_speech( $args = [], $opts = [] ) {
+	 *
+	 * [<post_ids>]
+	 * : Comma-delimited list of post IDs to generate excerpts for
+	 *
+	 * [--post_type=<post_type>]
+	 * : Batch process items belonging to this post type. If not used, relies on post_ids in args
+	 *
+	 * [--post_status=<post_status>]
+	 * : Batch process items that have this post status. Default publish
+
+	 * [--per_page=<int>]
+	 * : How many items should be processed at a time. Default 100
+	 *
+	 * [--force=<bool>]
+	 * : Whether to process items that already have an excerpt set. Default false
+	 *
+	 * [--dry-run=<bool>]
+	 * : Whether to run as a dry-run. Default true
+	 *
+	 * @param array $args Arguments.
+	 * @param array $opts Options.
+	 */
+	public function generate_excerpt( $args = [], $opts = [] ) {
+		$defaults = [
+			'post_type'   => false,
+			'post_status' => 'publish',
+			'per_page'    => 100,
+			'force'       => false,
+		];
+
+		$opts             = wp_parse_args( $opts, $defaults );
+		$opts['per_page'] = (int) $opts['per_page'] > 0 ? $opts['per_page'] : 100;
+
+		$count  = 0;
+		$errors = 0;
+
+		$chat_gpt = new ChatGPT( false );
+
+		// Determine if this is a dry run or not.
+		if ( isset( $opts['dry-run'] ) ) {
+			if ( 'false' === $opts['dry-run'] ) {
+				$dry_run = false;
+			} else {
+				$dry_run = (bool) $opts['dry-run'];
+			}
+		} else {
+			$dry_run = true;
+		}
+
+		if ( $dry_run ) {
+			\WP_CLI::line( '--- Running command in dry-run mode ---' );
+		}
+
+		// If we have a post type specified, process all items in that type.
+		if ( ! empty( $opts['post_type'] ) ) {
+			// Only allow processing post types that are enabled in settings.
+			if ( ! in_array( $opts['post_type'], get_post_types(), true ) ) {
+				\WP_CLI::error( sprintf( 'The "%s" post type is not a valid post type', $opts['post_type'] ) );
+			}
+
+			// Only allow processing post statuses that are valid for a particular post type.
+			if ( ! in_array( $opts['post_status'], get_available_post_statuses( $opts['post_type'] ), true ) ) {
+				\WP_CLI::error( sprintf( 'The "%s" post status is not valid for the "%s" post type', $opts['post_status'], $opts['post_type'] ) );
+			}
+
+			\WP_CLI::log( sprintf( 'Starting processing of "%s" post type items that have the "%s" status in batches of %d', $opts['post_type'], $opts['post_status'], $opts['per_page'] ) );
+
+			$paged = 1;
+
+			do {
+				$posts = get_posts(
+					array(
+						'post_type'        => $opts['post_type'],
+						'posts_per_page'   => $opts['per_page'],
+						'paged'            => $paged,
+						'post_status'      => $opts['post_status'],
+						'suppress_filters' => 'false',
+					)
+				);
+				$total = count( $posts );
+
+				foreach ( $posts as $post ) {
+					// Don't process if an item has an existing excerpt and we aren't forcing it.
+					if ( '' !== trim( $post->post_excerpt ) && ! $opts['force'] ) {
+						\WP_CLI::log( sprintf( 'Item ID %d has an existing excerpt and the force option hasn\'t been set. Skipping...', $post->ID ) );
+						$errors ++;
```

Comment:
Nitpick:
Maybe better to classify this as `skipped` and display something like `2 Items skipped` instead of grouped with errors `2 items had errors`?

Reply:
Makes sense, updated now